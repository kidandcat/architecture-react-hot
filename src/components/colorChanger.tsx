import * as React from "react";
import { Store } from "../state/store";
import { observer } from "mobx-react";
import { changeColor } from "../state/actions";
import { SyntheticEvent } from "react";
import { changeTempColor } from "../state/actions";

@observer
export class ColorChanger extends React.Component<{}, {}> {
  render() {
    return (
      <div>
        <input
          type="text"
          onChange={e => {
            changeTempColor(e.currentTarget.value);
          }}
          value={Store.color.tempValue}
        />
        <button onClick={() => changeColor(Store.color.tempValue)}>
          Change color
        </button>
      </div>
    );
  }
}
