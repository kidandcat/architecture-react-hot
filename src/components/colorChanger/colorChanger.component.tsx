import * as React from "react";
import { createStructuredSelector } from "reselect";
import { changeColor, changeTempColor } from "../../State/actions/color";
import { connect } from "react-redux";
import { selectTempColor } from "../../State/selectors";

export type ColorChangerProps = {
  changeColor: Function;
  changeTempColor: Function;
  temporalColor: string;
};

const mapStateToProps = createStructuredSelector({
  temporalColor: selectTempColor
});

const mapDispatchToProps = {
  changeColor,
  changeTempColor
};

@connect(mapStateToProps, mapDispatchToProps)
export class ColorChanger extends React.Component<ColorChangerProps> {
  render() {
    const { temporalColor, changeTempColor, changeColor } = this.props;
    return (
      <div>
        <input
          type="text"
          onChange={e => {
            changeTempColor(e.currentTarget.value);
          }}
          value={temporalColor}
        />
        <button onClick={() => changeColor(temporalColor)}>Change color</button>
      </div>
    );
  }
}
