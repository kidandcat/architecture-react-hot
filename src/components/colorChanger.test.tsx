import * as React from "react";
import { shallow } from "enzyme";

import { ColorChanger } from "./colorChanger";

test("<ColorChanger />", () => {
  const wrapper = shallow(<ColorChanger />);
  expect(wrapper).toMatchSnapshot();
});
