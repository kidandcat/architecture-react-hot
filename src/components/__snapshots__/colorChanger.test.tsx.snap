// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorChanger /> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorChanger />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          type="text"
          value="red"
        />,
        <button
          onClick={[Function]}
        >
          Change color
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "type": "text",
          "value": "red",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Change color",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Change color",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            type="text"
            value="red"
          />,
          <button
            onClick={[Function]}
          >
            Change color
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "text",
            "value": "red",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Change color",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Change color",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
